File: backend-py/seed.py
from database import engine, SessionLocal, Base
from models import User, Post, Like, Comment
from auth import hash_password

Base.metadata.create_all(bind=engine)

db = SessionLocal()

# Clear
db.query(Comment).delete()
db.query(Like).delete()
db.query(Post).delete()
db.query(User).delete()
db.commit()

pw = hash_password("password")

alice = User(username="alice", password_hash=pw, avatar_url="https://i.pravatar.cc/150?u=alice")
bob = User(username="bob", password_hash=pw, avatar_url="https://i.pravatar.cc/150?u=bob")
charlie = User(username="charlie", password_hash=pw, avatar_url="https://i.pravatar.cc/150?u=charlie")
db.add_all([alice, bob, charlie])
db.commit()
db.refresh(alice); db.refresh(bob); db.refresh(charlie)

post1 = Post(text="Hello world! This is my first post 🌍", author_id=alice.id)
post2 = Post(text="Beautiful sunset today 🌅", image_url="https://picsum.photos/seed/sunset/800/600", author_id=bob.id)
post3 = Post(text="Just shipped a new feature! 🚀", author_id=charlie.id)
db.add_all([post1, post2, post3])
db.commit()
db.refresh(post1); db.refresh(post2); db.refresh(post3)

db.add_all([
    Like(post_id=post1.id, user_id=bob.id),
    Like(post_id=post1.id, user_id=charlie.id),
    Like(post_id=post2.id, user_id=alice.id),
])
db.commit()

db.add_all([
    Comment(text="Welcome! 🎉", post_id=post1.id, author_id=bob.id),
    Comment(text="Amazing photo!", post_id=post2.id, author_id=alice.id),
    Comment(text="Congrats! 🎊", post_id=post3.id, author_id=alice.id),
    Comment(text="Well done!", post_id=post3.id, author_id=bob.id),
])
db.commit()

print("Seeded: 3 users, 3 posts, 3 likes, 4 comments")
print("Login with username: alice/bob/charlie, password: password")
db.close()

File: backend-js/src/seed.ts
import prisma from './prisma';
import bcrypt from 'bcryptjs';

async function seed() {
  console.log('Seeding database...');

  await prisma.comment.deleteMany();
  await prisma.like.deleteMany();
  await prisma.post.deleteMany();
  await prisma.user.deleteMany();

  const hash = await bcrypt.hash('password', 10);

  const alice = await prisma.user.create({
    data: { username: 'alice', passwordHash: hash, avatarUrl: 'https://i.pravatar.cc/150?u=alice' },
  });
  const bob = await prisma.user.create({
    data: { username: 'bob', passwordHash: hash, avatarUrl: 'https://i.pravatar.cc/150?u=bob' },
  });
  const charlie = await prisma.user.create({
    data: { username: 'charlie', passwordHash: hash, avatarUrl: 'https://i.pravatar.cc/150?u=charlie' },
  });

  const post1 = await prisma.post.create({
    data: { text: 'Hello world! This is my first post 🌍', authorId: alice.id },
  });
  const post2 = await prisma.post.create({
    data: { text: 'Beautiful sunset today 🌅', imageUrl: 'https://picsum.photos/seed/sunset/800/600', authorId: bob.id },
  });
  const post3 = await prisma.post.create({
    data: { text: 'Just shipped a new feature! 🚀', authorId: charlie.id },
  });

  await prisma.like.create({ data: { postId: post1.id, userId: bob.id } });
  await prisma.like.create({ data: { postId: post1.id, userId: charlie.id } });
  await prisma.like.create({ data: { postId: post2.id, userId: alice.id } });

  await prisma.comment.create({ data: { text: 'Welcome! 🎉', postId: post1.id, authorId: bob.id } });
  await prisma.comment.create({ data: { text: 'Amazing photo!', postId: post2.id, authorId: alice.id } });
  await prisma.comment.create({ data: { text: 'Congrats! 🎊', postId: post3.id, authorId: alice.id } });
  await prisma.comment.create({ data: { text: 'Well done!', postId: post3.id, authorId: bob.id } });

  console.log('Seeded: 3 users, 3 posts, 3 likes, 4 comments');
  console.log('Login with username: alice/bob/charlie, password: password');
}

seed()
  .catch(console.error)
  .finally(() => prisma.$disconnect());
